### PR TITLE
Adds guidance on how to test transform configuration/validation

### PR DESCRIPTION
**Pull request by jkff**
R: @tgroh

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/beam_PreCommit_Website_Stage/650/

Jenkins built the site at commit id 5eed3c813be56f60961000feab7b93b699b69fda with Jekyll and staged it [here](http://apache-beam-website-pull-requests.storage.googleapis.com/301/index.html). Happy reviewing.

Note that any previous site has been deleted. This staged site will be automatically deleted after its TTL expires. Push any commit to the pull request branch or re-trigger the build to get it staged again.

**Comment by kennknowles** [comment]
R: @melap (feel free to decline, just sending things your way if I think they might be in your wheelhouse)

File: src/contribute/ptransform-style-guide.md [hunk 1]
```diff
@@ -162,11 +164,25 @@ Data processing is tricky, full of corner cases, and difficult to debug, because
...
 * For detailed instructions on test code organization, see the [Beam Testing Guide]({{ site.baseurl }}/contribute/testing/).
 
+#### Testing transform construction and validation
+
+The code for constructing and validating a transform is usually trivial and mostly boilerplate. However, minor mistakes or typos in it can have serious consequences (e.g. ignoring a property that the user has set), so it needs to be tested as well. Yet, an excessive amount of trivial tests can be hard to maintain and give a false impression that the transform is well-tested.
+
+Do:
+* Test non-trivial validation code: where missing/incorrect/uninformative validation may lead to serious problems: data loss, counter-intuitive behavior, value of a property being silently ignored, or other hard-to-debug errors. Create 1 test per non-trivial class of validation error. Some examples of validation that should be tested:
```
Comment (melap) [inline src/contribute/ptransform-style-guide.md]: nit - the : after validation code: seems extraneous

**Comment by jkff** [comment]
Thomas is OOO so, say, R: @robertwb

**Comment by robertwb** [comment]
LGTM, thanks.

**Comment by jkff** [comment]
@asfgit merge

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/beam_PreCommit_Website_Stage/661/

Jenkins built the site at commit id 09ea01eae10366137ce0c5000d104f391bf57220 with Jekyll and staged it [here](http://apache-beam-website-pull-requests.storage.googleapis.com/301/index.html). Happy reviewing.

Note that any previous site has been deleted. This staged site will be automatically deleted after its TTL expires. Push any commit to the pull request branch or re-trigger the build to get it staged again.
